### PR TITLE
added 404 page

### PR DESCRIPTION
**Pull request by SarthakC**
added 404 page
updated route for 404 page
added font awesome
beautified header using font awesome
refactored signUpToggle function

## PR Checklist
Please check that your PR fulfills the following requirements:

- [x] The commit message follows [our guidelines](https://github.com/pesto-students/batch-10-tripen/wiki/Commit-message-guidelines)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] GIFs/Videos/Images added (compulsory for frontend changes)

## PR Type
What kind of change does this PR introduce? 

- [ ] Bugfix
- [x] Feature
- [ ] Tests changed / added / updated
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Infrastructure changes
- [ ] Dependency upgrade
- [ ] Other... Please describe:

## Issue Number: 
#91 

## What is the new behavior?
Redirects user to a dedicated page in case of a 404 error. 

## Does this PR introduce a breaking change?
- [ ] Yes
- [ ] No

### CODE REVIEW

File: client/src/components/organisms/Header/Header.jsx [hunk 1]
```diff
@@ -1,19 +1,28 @@
 import React from 'react';
 import { Navbar, Nav, Button } from 'react-bootstrap';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
```
Comment (p-racoon) [inline client/src/components/organisms/Header/Header.jsx]: Could it have been achieved just by using CDN links

File: client/src/components/organisms/Header/Header.jsx [hunk 2]
```diff
@@ -1,19 +1,28 @@
...
+import { faRoute, faPlus } from '@fortawesome/free-solid-svg-icons';
```
Comment (p-racoon) [inline client/src/components/organisms/Header/Header.jsx]: Could have been seperate PR

File: client/src/components/organisms/Header/Header.jsx [hunk 3]
```diff
@@ -1,19 +1,28 @@
...
-        </Nav>
-      </Navbar.Collapse>
-    </Navbar>
-  );
-};
+const Header = () => (
+  <Navbar bg="light" expand="lg">
+    <Navbar.Brand href="/">
+      <FontAwesomeIcon icon={faRoute} />
+      {' '}
```
Comment (karansapolia) [inline client/src/components/organisms/Header/Header.jsx]: Why this solitary space?

File: client/src/components/organisms/Header/Header.jsx [hunk 4]
```diff
@@ -1,19 +1,28 @@
...
+      Tripen
```
Comment (karansapolia) [inline client/src/components/organisms/Header/Header.jsx]: I think now is a good time to replace 'Tripen' with a logo (component). Maybe just place in the react logo from CRA for now.

File: client/src/components/organisms/Header/Header.jsx [hunk 5]
```diff
@@ -1,19 +1,28 @@
...
+    </Navbar.Brand>
+    <Navbar.Toggle aria-controls="basic-navbar-nav" />
+    <Navbar.Collapse id="basic-navbar-nav">
+      <Nav className="ml-auto">
+        <Button variant="primary">
+          <FontAwesomeIcon icon={faPlus} />
+          {' '}
```
Comment (karansapolia) [inline client/src/components/organisms/Header/Header.jsx]: remove this

File: client/src/components/organisms/LoginModal/LoginModal.jsx [hunk 6]
```diff
@@ -9,21 +12,21 @@ const LoginModal = () => {
...
-    signUp ? initSignIn() : initSignUp();
+    setSignUp(!signUp);
   };
 
   return (
     <>
-      <Button variant='outline-secondary' onClick={handleShow}>
+      <Button variant="outline-secondary" onClick={handleShow}>
+        <FontAwesomeIcon icon={faSignInAlt} />
+        {' '}
```
Comment (karansapolia) [inline client/src/components/organisms/LoginModal/LoginModal.jsx]: remove all such spaces

File: client/src/components/pages/NotFound/NotFound.jsx [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+const NotFound = () => (
+  <Container className="text-center mt-5">
+    <h1>Oops!</h1>
+    <h1>404 Not Found</h1>
+    <p>Sorry, an error has occured, Requested page not found!</p>
+    <Link to="/">
+      <Button>
+        <FontAwesomeIcon icon={faHome} />
+        {' '}
```
Comment (karansapolia) [inline client/src/components/pages/NotFound/NotFound.jsx]: remove

File: client/src/components/organisms/Header/Header.jsx (hunk 1 above)
Comment (SarthakC) [reply client/src/components/organisms/Header/Header.jsx]: I agree. My bad

File: client/src/components/organisms/Header/Header.jsx (hunk 3 above)
Comment (SarthakC) [reply client/src/components/organisms/Header/Header.jsx]: The space is to improve the aesthetics of the Button.

File: client/package.json [hunk 8]
```diff
@@ -3,6 +3,11 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@babel/core": "^7.0.0-0",
```
Comment (p-racoon) [inline client/package.json]: Do we need it?

File: client/package.json [hunk 9]
```diff
@@ -3,6 +3,11 @@
...
+    "@fortawesome/fontawesome-svg-core": "^1.2.25",
+    "@fortawesome/free-solid-svg-icons": "^5.11.2",
+    "@fortawesome/react-fontawesome": "^0.1.4",
+    "babel-loader": "^8.0.6",
```
Comment (p-racoon) [inline client/package.json]: Is it the one that Arfat told us to correct?

File: client/src/components/pages/NotFound/NotFound.jsx [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+import { Link } from 'react-router-dom';
+import { Button, Container } from 'react-bootstrap';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faHome } from '@fortawesome/free-solid-svg-icons';
+
+
+const NotFound = () => (
+  <Container className="text-center mt-5">
+    <h1>Oops!</h1>
+    <h1>404 Not Found</h1>
```
Comment (p-racoon) [inline client/src/components/pages/NotFound/NotFound.jsx]: Would it Always be 404 error?

**Comment by SarthakC** [comment]
> Attach GIFs/Videos for frontend PRs..

Added.

File: client/package.json (hunk 9 above)
Comment (SarthakC) [reply client/package.json]: no not this one

**Pull request merged**